File: src/sources/cdc.ts
import Parser from 'rss-parser';
import { BaseSource, RawArticle } from './base';
import { logger } from '@/lib/logger';

const parser = new Parser({ timeout: 15000, maxRedirects: 5 });

const FEEDS = [
  'https://tools.cdc.gov/api/v2/resources/media/403372.rss',
  'https://emergency.cdc.gov/han/rss.asp',
  'https://www.cdc.gov/mmwr/feeds/rss/wk.xml',
];

export class CDCSource extends BaseSource {
  readonly name = 'CDC';

  async fetch(): Promise<RawArticle[]> {
    const articles: RawArticle[] = [];
    for (const feedUrl of FEEDS) {
      try {
        const feed = await parser.parseURL(feedUrl);
        for (const item of feed.items) {
          if (!item.title || !item.link) continue;
          articles.push({
            source: this.name,
            title: this.stripHtml(item.title),
            url: item.link,
            published_at: this.parseDate(item.pubDate ?? item.isoDate),
            summary: item.contentSnippet ?? '',
            language: this.language,
          });
        }
      } catch (e) {
        logger.warn(`[CDC] 피드 오류 ${feedUrl}: ${e}`);
      }
    }
    return articles;
  }
}

File: src/sources/cdc.py
import feedparser
from .base import BaseSource, RawArticle


class CDCSource(BaseSource):
    name = "CDC"
    language = "en"
    FEEDS = [
        "https://tools.cdc.gov/api/v2/resources/media/403372.rss",  # CDC Newsroom
        "https://emergency.cdc.gov/han/rss.asp",  # Health Alert Network
        "https://www.cdc.gov/mmwr/feeds/rss/wk.xml",  # MMWR Weekly
    ]

    async def fetch(self) -> list[RawArticle]:
        articles = []
        for url in self.FEEDS:
            try:
                feed = feedparser.parse(url)
                for entry in feed.entries:
                    articles.append(RawArticle(
                        source=self.name,
                        title=entry.get("title", "").strip(),
                        url=entry.get("link", ""),
                        published_at=self._safe_date(entry.get("published")),
                        summary=entry.get("summary", ""),
                        language=self.language,
                    ))
            except Exception as e:
                from src.utils.logger import logger
                logger.warning(f"[CDC] 피드 오류 {url}: {e}")
        return articles
